src/features/tokens/TokenListModal.tsx

```diff
@@ -82,75 +82,81 @@ export function TokenList({
 }) {
   const tokens = useMemo(() => {
     const q = searchQuery?.trim().toLowerCase();
-    return getTokens()
-      .map((t) => {
-        const hasRoute = hasTokenRoute(
-          originCaip2Id,
-          destinationCaip2Id,
-          t.tokenCaip19Id,
-          tokenRoutes,
-        );
-        return { ...t, disabled: !hasRoute };
-      })
-      .sort((a, b) => {
-        if (a.disabled && !b.disabled) return 1;
-        else if (!a.disabled && b.disabled) return -1;
-        else return 0;
-      })
-      .filter((t) => {
-        if (!q) return t;
-        return (
-          t.name.toLowerCase().includes(q) ||
-          t.symbol.toLowerCase().includes(q) ||
-          t.tokenCaip19Id.toLowerCase().includes(q)
-        );
-      })
-      .filter((t) => (config.showDisabledTokens ? true : !t.disabled));
+    return (
+      getTokens()
+        .map((t) => {
+          const hasRoute = hasTokenRoute(
+            originCaip2Id,
+            destinationCaip2Id,
+            t.tokenCaip19Id,
+            tokenRoutes,
+          );
+          return { ...t, disabled: !hasRoute };
+        })
+        .sort((a, b) => {
+          if (a.disabled && !b.disabled) return 1;
+          else if (!a.disabled && b.disabled) return -1;
+          else return 0;
+        })
+        // Remove duplicates
+        .filter((t, i, list) => i === list.findIndex((t2) => t2.tokenCaip19Id === t.tokenCaip19Id))
+        // Filter down to search query
+        .filter((t) => {
+          if (!q) return t;
+          return (
+            t.name.toLowerCase().includes(q) ||
+            t.symbol.toLowerCase().includes(q) ||
+            t.tokenCaip19Id.toLowerCase().includes(q)
+          );
+        })
```

Comment:
not sure why the diff is rendering so much

Reply:
Yeah strange